crates/astria-sequencer/src/app.rs

```diff
@@ -523,6 +523,70 @@ mod test {
         );
     }
 
+    #[tokio::test]
+    async fn app_deliver_tx_invalid_nonce() {
+        // Arrange
+        // initialize app from genesis state with default_genesis_accounts
+        // (balances are 10u128.pow(19))
+        let storage = penumbra_storage::TempStorage::new()
+            .await
+            .expect("failed to create temp storage backing chain state");
+        let snapshot = storage.latest_snapshot();
+        let mut app = App::new(snapshot);
+        let genesis_state = GenesisState {
+            accounts: default_genesis_accounts(),
+        };
+        app.init_chain(genesis_state).await.unwrap();
+
+        // initialize tx signer
+        // this secret key corresponds to ALICE_ADDRESS
+        let alice_secret_bytes: [u8; 32] =
+            hex::decode("2bd806c97f0e00af1a1fc3328fa763a9269723c8db8fac4f93af71db186d6e90")
+                .unwrap()
+                .try_into()
+                .unwrap();
+        let alice_signing_key = SigningKey::from(alice_secret_bytes);
+        let alice = Address::from_verification_key(alice_signing_key.verification_key());
+
+        // create tx with invalid nonce 1
+        let data = b"hello world".to_vec();
+        let tx = UnsignedTransaction {
+            nonce: 1,
+            actions: vec![
+                SequenceAction {
+                    chain_id: b"testchainid".to_vec(),
+                    data,
+                }
+                .into(),
+            ],
+        };
+
+        // sign tx
+        let signed_tx = tx.into_signed(&alice_signing_key);
+        let bytes = signed_tx.into_raw().encode_to_vec();
+
+        // Act
+        // pass tx to deliver_tx
+        let response = app.deliver_tx(&bytes).await;
```

Comment:
if you want to keep this unit test, i'd move this code into its own function (as it's essentially the same as the other tests) that takes an `UnsignedTransaction` and returns the response